src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -313,16 +326,21 @@ public void addHooks(Hook... hooks) {
      * @return A list of {@link Hook}s.
      */
     public List<Hook> getHooks() {
-        return new ArrayList<>(this.apiHooks);
+        // Hooks can be duplicated if they support multiple FlagValueTypes
```

Comment:
I am not sure, the order of hooks is important, i know that you are now using other methods, to retrieve the needed hooks, but can this implementation now be a problem, because we are not keeping the order of the hooks. Now we merge multiple hook source after each other.